src/AzureAuth/CommandMain.cs

```diff
@@ -218,6 +222,46 @@ public static string PrefixedPromptHint(string promptHint)
             return $"{PromptHintPrefix}: {promptHint}";
         }
 
+        /// <summary>
+        /// Generates event data from the AuthFlowResult.
+        /// </summary>
+        /// <param name="result">The AuthFlowResult.</param>
+        /// <returns>The event data.</returns>
+        public EventData GenerateEventData(AuthFlowResult result)
+        {
+            if (result == null)
+            {
+                return null;
+            }
+
+            var eventData = new EventData();
+            eventData.Add("authflow", result.AuthFlowName);
+            eventData.Add("success", result.Success);
+
+            var correlationIDs = new List<string>();
+
+            if (result.Errors.Any())
+            {
+                var error_messages = JsonSerializer.Serialize(result.Errors.Select(error => ExceptionsExtensions.ToFormattedString(error)));

```

Comment:
I don't think we need JSON serialization here. We can use the existing ExceptionListToStringConverter class and method without modiciation to turn this list of errors, into a collecting string (with 1 error message per line) which is easy to parse in Kusto.